modules/saml/www/proxy/invalid_session.php

```diff
@@ -16,10 +16,18 @@
 
 try {
     // try to get the state
+    /** @var array $state */
     $state = \SimpleSAML\Auth\State::loadState($_REQUEST['AuthState'], 'saml:proxy:invalid_idp');
+    if ($state === null) {
+        throw new \SimpleSAML\Error\NoState();
+    }
 } catch (\Exception $e) {
     // the user probably hit the back button after starting the logout, try to recover the state with another stage
+    /** @var array $state */
     $state = \SimpleSAML\Auth\State::loadState($_REQUEST['AuthState'], 'core:Logout:afterbridge');
+    if ($state === null) {
+        throw new \SimpleSAML\Error\NoState();
+    }
```

Comment:
Can probably do without the annotation

Reply:
That's true, psalm should be smart enough to notice that...

Reply:
Annotations removed

Reply:
Actually, looking into `loadState()`, since we're not passing `true` to the third argument (`$allowMissing`), if we cannot find the state, we'll either redirect or thrown an exception, meaning `$state` can never be `null` and then these checks are superfluous...

Reply:
Oh good catch! I totally missed that!